Fixup Deep Sleep in combination with bluesleep.

Diewi opened the pull request:
Disable the wake notification via the procfs, since it performed by
issuing an ioctl now.

Change-Id: I3e19167ab3b478af8f5cfcb297bed94d3489792e
Signed-off-by: Alexander Diewald <Diewi@diewald-net.com>

humberos commented:
I am going to go test this patch (thanks btw) but first I'd think that we can drop out BT_WAKE_VIA_PROC_NOTIFY_DEASSERT since it depends on BT_WAKE_VIA_PROC .

humberos commented:
Better, we can drop both since they are FALSE by default.
https://github.com/omnirom/android_hardware_broadcom_libbt/blob/android-7.1/include/bt_vendor_brcm.h#L213-L223

humberos commented:
Also PROC_BTWRITE_TIMER_TIMEOUT_MS  so that's the list:

- BT_WAKE_VIA_PROC = FALSE
- BT_WAKE_VIA_PROC_NOTIFY_DEASSERT = FALSE
- PROC_BTWRITE_TIMER_TIMEOUT_MS = 0

humberos commented:
remove all

Diewi commented:
Thanks for looking into this. I will remove these lines as you noted (I was unsure whether to keep them as reference or not).

I found that these defines are also equivalent to the defaults:
- UART_TARGET_BAUD_RATE
- SCO_USE_I2S_INTERFACE 

Here, I'd provide second cleanup commit since removing these lines does not relate to the purpose of this commit. What do you think?

Diewi commented:
Updated the commit (removed the three proc-related lines).

Also, I have updated the commits for loire and kitakami.

humberos commented:
Please, do a simple test.
Turn ON the bluetooth and restart your device.
After a complete restart the bluetooth should be ON still, then ok leave your device for 3 minutes (black screen) and check deep sleep state.
Deep sleep state should also be ok no wakelocks so fine so good but.... now this is the crucial moment 
Just turn OFF bluetooth.... then.... you will get bluesleep wakelock and also broken deep sleep.

Diewi commented:
Just followed the procedure you described, and I do not get the bluesleep wakelock. However, I will re-try and monitor this the next hours. Maybe, the device gets into some inconsistent state.

Just to be sure: I assume you have applied the kernel commit as well (I am pretty sure you did since deep sleep should be completely broken otherwise)?

rcstar6696 commented:
yes humberto is right
 after turn off bluetooth the wakelock is there again

nailyk-fr commented:
I'm on patchset 1 and, with bluetooth on, wakelocks are present:
http://pix.toile-libre.org/upload/original/1483122345.png
(LineageOS)

Diewi commented:
@nailyk-fr: @rcstar6696 could reproduce what you described. I will examine this issue closer tomorrow. I could not repoduce the event wakelock problem up to now.
Which device and BT config do you have (connected devices & paired devices)? This might be related.

Diewi commented:
@humberos @nailyk-fr @rcstar6696 

I could reproduce the event wakelock issue when BT is turned OFF now with a fresh android installation with no BT devices paired.  

I also have updated the kernel patch that solves the problem on my device (tested both with no paired devices and paired devices using the procedure described by humberto). In essence, the bluesleep_start/stop functions were called from the wrong places.

jerpelea commented:
now it looks better but we have tobtest it

nailyk-fr commented:
Thanks.
Here is my experiment with the new shinano PS and kernel#1095:
[1. BT off](http://pix.toile-libre.org/upload/original/1483196448.png)
[2. BT on not paired](http://pix.toile-libre.org/upload/original/1483196487.png)
[3. reboot then BT on & paired](http://pix.toile-libre.org/upload/original/1483196516.png)
[4. BT turned off without proper unpair. ](http://pix.toile-libre.org/upload/original/1483197624.png)

I can provide any log you want. 
(still LineageOS, with leo/d6603)

Diewi commented:
Thanks for testing it @nailyk-fr,

To get things right: In which cases does your device go into deep sleep? I'd assume 1 and 2 (and possibly 3)?
It would be nice if you could provide me a dmesg for the cases where it does not work, 2-3 minutes after bootup when the devices tries to go into deep sleep mode.

nailyk-fr commented:
Thanks you for fixing this :)
Sure will do that asap. 
Yes, at 1 & 2 the device sleep fine. 
At 3 & 4 no more sleep. I don't remember exactly but it seems the wakelock is not the same as before (bluesleep before, now event[1-4]) will confirm.

nailyk-fr commented:
**Step to reproduce**
1a. pair computer, 
2a. reboot device (BT on), 
3a. computer is auto connected, 
4a. play audio trough a2dp, 
5a. turn screen off wait a bit, turn screen on, 
6a. Turn BT off, 
7a. switch screen off.
  *Device doesn't sleep anymore.*
Full [dmesg.zip](https://github.com/sonyxperiadev/device-sony-shinano/files/679730/dmesg.zip) (Error is repeated each second)

**Sleep working**
1b. pair computer, 
2b. reboot with bluetooth on, 
3b. computer is auto connected, 
4b. turn bt off, 
5b. switch screen off
  *Sleep ok.*

Diewi commented:
@nailyk-fr 
Thanks for the detailed report and the log file. I will look into it tomorrow.

Meanwhile, I updated the kernel commit since I missed something obvious (see the PR). I think the mistake that I corrected there could be the cause of the behaviour you are observing: broken pairing causeing event wakelocks etc. Maybe you want to test the updated patch. :)

The problem was that the bluesleep driver was not notified of outgoing data by the serial driver, which caused the sleep code to kick in while uart was still active. Also, data transmission was somewhat broken, but looks good now.

nailyk-fr commented:
Perfect now :) Thanks for patch and reactivity. 
Are other tests needed? 

btw: I though the music broke when screen off because my motherboard miss the antenna, but now music play even if the screen is dark.

Diewi commented:
Ok, thanks. Nice to hear.

I t would be great if you could run your phone with the patch applied for some days and report any unexpected behaviour here. Just for coverage :)
